src/evm/engine_db/simulation_db.rs

```diff
@@ -105,12 +98,9 @@ pub struct SimulationDB<M: Middleware + Debug> {
     pub runtime: Option<Arc<tokio::runtime::Runtime>>,
 }
 
-impl<M: Middleware + Debug> SimulationDB<M>
-where
-    M::Error: std::fmt::Debug,
-{
+impl<P: Provider + Debug + 'static> SimulationDB<P> {
```

Comment:
Why do we need it to be `'static`?

Reply:
if I don't, I get

```
error[E0310]: the parameter type `P` may not live long enough
   --> src/evm/engine_db/simulation_db.rs:270:55
    |
270 | impl<P: Provider + Debug> EngineDatabaseInterface for SimulationDB<P>
    |                                                       ^^^^^^^^^^^^^^^
    |                                                       |
    |                                                       the parameter type `P` must be valid for the static lifetime...
    |                                                       ...so that the type `P` will meet its required lifetime bounds...
    |
note: ...that is required by this bound
   --> src/evm/engine_db/engine_db_interface.rs:6:36
    |
6   | pub trait EngineDatabaseInterface: DatabaseRef + Send + Sync {
    |                                    ^^^^^^^^^^^
help: consider adding an explicit lifetime bound
    |
270 | impl<P: Provider + Debug + 'static> EngineDatabaseInterface for SimulationDB<P>
    |                          +++++++++
```